packages/forge/src/governance/extensions/GovernorSorting.sol

```diff
@@ -2,232 +2,189 @@
 
 pragma solidity ^0.8.0;
 
-import "@openzeppelin/utils/math/SafeCast.sol";
-import "./GovernorCountingSimple.sol";
-
 /**
- * @dev Extension of {GovernorCountingSimple} for sorting and ranking.
- *
- * _Available since v4.3._
+ * @dev Logic for sorting and ranking.
  */
-abstract contract GovernorSorting is GovernorCountingSimple {
-    bool public setSortedAndTiedProposalsHasBeenRun = false;
-    mapping(uint256 => uint256) public tiedAdjustedRankingPosition; // key is ranking, value is index of the last iteration of that ranking's value in the _sortedProposalIds array taking ties into account
-
-    mapping(uint256 => bool) private _isTied; // whether a ranking is tied. key is ranking.
-    uint256[] private _sortedProposalIds;
-    uint256 private _lowestRanking; // worst ranking (1 is the best possible ranking, 8 is a lower/worse ranking than 1)
-    uint256 private _highestTiedRanking; // best (1 is better than 8) ranking that is tied
+abstract contract GovernorSorting {
+    // Because of the array rule below, the actual number of rankings that this contract will be able to track is determined by three things:
+    //      - RANK_LIMIT
+    //      - Woulda Beens (WBs)
+    //          The number of would-be ranked proposals (at the end of the contest if rankings were counted
+    //          without taking out deleted proposals) within the limit that are deleted and do not have other, non-deleted proposals
+    //          with the same amounts of votes/that are tied with them.
+    //      - To Tied (TTs)
+    //          The number of times a proposal is voted into a ranking to tie it or a ranking that is already tied
+    //          from a ranking that was in the tracked rankings at the time that vote was cast.
+    //
+    // The equation to calcluate how many rankings this contract will actually be able to track is:
+    // # of rankings GovernorSorting can track for a given contest = RANK_LIMIT - WBs - TTs
+    //
+    // With this in mind, it is strongly reccomended to set RANK_LIMIT sufficiently high to create a buffer for
+    // WBs and TTs that may occur in your contest. The thing to consider with regard to making it too high is just
+    // that it is more gas for users on average the higher that RANK_LIMIT is set.
+
+    // TODO: add option to disable rank tracking for gas savings
+    // TODO: make RANK_LIMIT configurable + test ranges
+    // TODO: flesh out tests a ton for different edge cases w oldValue, smallestNonZeroSortedRanksValueIdx, and RANK_LIMIT overlaps
+    uint256 public constant RANK_LIMIT = 250; // cannot be 0
+
+    // RULE: array length can never end lower than it started a transaction, otherwise erroneous ranking can happen
+    uint256[] public sortedRanks = new uint256[](RANK_LIMIT); // value is forVotes counts, has the constraint of no duplicate values except for 0 bc it's instantiated with 0 values.
```

Comment:
I'm curious: why did you choose to have a fixed size array here? I understand that you wanted a limit so this array cannot become infinitely large, but I feel like this comes with some inconveniences and it might be more error prone - if you used a non-fixed-size array, one optimization could be to use it's push method to push a value to it's smallest non-zero index (this way you wouldn't have to track indexes separately.
If it's because of gas usage (you copy the array to storage multiple times) you still might want to reconsider this as you could have a simple for loop and copy elements one by one - this might be even cheaper if not all the 250 values are populated in the array-

Reply:
Fantastic call, yeah this makes the code much cleaner and also means I don't have to explicitly track a `smallestValueIdx` variable anymore, which removes possible error surface area, thank you!